git-hooks/pre-commit.sh

```diff
@@ -14,7 +15,7 @@ yellow='\033[1;33m'
 # carriage return hack. Leave it on 2 lines.
 cr='
 '
-for f in $(git diff --cached --name-only | grep -E "$FILES_PATTERN")
+for f in $(git diff --cached --name-only | grep -E "FILES_PATTERN_CONTAIN" | grep -E "FILES_PATTERN_NOT_CONTAIN")
```

Comment:
You'll need dollar-signs here c.f. `$FILES_PATTERN`.

Reply:
You'll also need `-v` to exclude in the second grep

Reply:
Does the -v say "don't match on this"? If so, I might not need that since the regex pattern I used is to exclude strings:
https://stackoverflow.com/questions/1395177/regex-to-exclude-a-specific-string-constant

Reply:
> Is that just how you reference variables in shell scripts?
Yup! And in the (bash) command line in general. (Pretty much anything you can do in a shell script you can also do in the command line and vice versa.)

Reply:
Cool, did that here: https://github.com/dimagi/commcare-cloud/pull/1762/commits/8e751ace1aaa8ea00ebb60e4f01d2b6d3f8fb5bc